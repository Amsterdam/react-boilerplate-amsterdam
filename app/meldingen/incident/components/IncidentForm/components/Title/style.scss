.title {
  &__subtitle {
    color: #999;
  }

  &__label {
  }
}
