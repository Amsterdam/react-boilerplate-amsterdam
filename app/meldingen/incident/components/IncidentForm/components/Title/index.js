import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const Title = ({ meta }) => (
  <div className=" col-12 title">
    <div className="title__label label">{meta.label}</div>
    <div className="title__subtitle">{meta.subtitle}</div>
  </div>
);

Title.propTypes = {
  meta: PropTypes.object.isRequired
};

export default Title;
