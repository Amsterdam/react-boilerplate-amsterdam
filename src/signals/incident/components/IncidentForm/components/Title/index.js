import React from 'react';
import PropTypes from 'prop-types';

import { Validators } from 'react-reactive-form';

import './style.scss';

const Title = ({ meta, options }) => (
  <div className="title">
    <div className="title__label">{meta && meta.label}
      {!options || !options.validators || (options && options.validators && !options.validators.includes(Validators.required)) ?
        <span className="title--not-required">(niet verplicht)</span>
      : ''}
    </div>
    <div className="title__subtitle">{meta && meta.subtitle}</div>
  </div>
);

Title.propTypes = {
  meta: PropTypes.object,
  options: PropTypes.object
};

export default Title;
