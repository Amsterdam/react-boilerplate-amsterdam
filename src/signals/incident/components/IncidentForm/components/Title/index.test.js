import React from 'react';
import { shallow } from 'enzyme';

import Title from './index';

describe('Form component <Title />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Title />);
  });

  it('should render correctly', () => {
    wrapper.setProps({
      meta: {
        label: 'Title'
      }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
