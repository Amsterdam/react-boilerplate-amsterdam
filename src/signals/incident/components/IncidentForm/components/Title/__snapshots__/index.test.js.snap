// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component <Title /> should render correctly 1`] = `
<div
  className="title"
>
  <div
    className="title__label"
  >
    Title
    <span
      className="title--not-required"
    >
      (niet verplicht)
    </span>
  </div>
  <div
    className="title__subtitle"
  />
</div>
`;
